commit 4cb1fc6fffe4910845e183d1a2dfe9509ba1062c
Merge: 740afa4d3941 aaa4dd1b47f5
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Tue Dec 13 15:22:14 2022 -0800

    Merge tag 'for-linus' of git://git.armlinux.org.uk/~rmk/linux-arm
    
    Pull ARM updates from Russell King:
    
     - update unwinder to cope with module PLTs
    
     - enable UBSAN on ARM
    
     - improve kernel fault message
    
     - update UEFI runtime page tables dump
    
     - avoid clang's __aeabi_uldivmod generated in NWFPE code
    
     - disable FIQs on CPU shutdown paths
    
     - update XOR register usage
    
     - a number of build updates (using .arch, thread pointer, removal of
       lazy evaluation in Makefile)
    
     - conversion of stacktrace code to stackwalk
    
     - findbit assembly updates
    
     - hwcap feature updates for ARMv8 CPUs
    
     - instruction dump updates for big-endian platforms
    
     - support for function error injection
    
    * tag 'for-linus' of git://git.armlinux.org.uk/~rmk/linux-arm: (31 commits)
      ARM: 9279/1: support function error injection
      ARM: 9277/1: Make the dumped instructions are consistent with the disassembled ones
      ARM: 9276/1: Refactor dump_instr()
      ARM: 9275/1: Drop '-mthumb' from AFLAGS_ISA
      ARM: 9274/1: Add hwcap for Speculative Store Bypassing Safe
      ARM: 9273/1: Add hwcap for Speculation Barrier(SB)
      ARM: 9272/1: vfp: Add hwcap for FEAT_AA32I8MM
      ARM: 9271/1: vfp: Add hwcap for FEAT_AA32BF16
      ARM: 9270/1: vfp: Add hwcap for FEAT_FHM
      ARM: 9269/1: vfp: Add hwcap for FEAT_DotProd
      ARM: 9268/1: vfp: Add hwcap FPHP and ASIMDHP for FEAT_FP16
      ARM: 9267/1: Define Armv8 registers in AArch32 state
      ARM: findbit: add unwinder information
      ARM: findbit: operate by words
      ARM: findbit: convert to macros
      ARM: findbit: provide more efficient ARMv7 implementation
      ARM: findbit: document ARMv5 bit offset calculation
      ARM: 9259/1: stacktrace: Convert stacktrace to generic ARCH_STACKWALK
      ARM: 9258/1: stacktrace: Make stack walk callback consistent with generic code
      ARM: 9265/1: pass -march= only to compiler
      ...

diff --cc arch/arm/Kconfig
index a08c9d092a33,981bbfa62b7d..65d0daf63262
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@@ -27,7 -28,9 +28,8 @@@ config AR
  	select ARCH_HAVE_NMI_SAFE_CMPXCHG if CPU_V7 || CPU_V7M || CPU_V6K
  	select ARCH_HAS_GCOV_PROFILE_ALL
  	select ARCH_KEEP_MEMBLOCK
+ 	select ARCH_HAS_UBSAN_SANITIZE_ALL
  	select ARCH_MIGHT_HAVE_PC_PARPORT
 -	select ARCH_NO_SG_CHAIN if !ARM_HAS_SG_CHAIN
  	select ARCH_OPTIONAL_KERNEL_RWX if ARCH_HAS_STRICT_KERNEL_RWX
  	select ARCH_OPTIONAL_KERNEL_RWX_DEFAULT if CPU_V7
  	select ARCH_SUPPORTS_ATOMIC_RMW